host-controller/src/main/java/org/jboss/as/host/controller/HostControllerService.java

```diff
@@ -128,6 +130,20 @@ public void start(StartContext context) throws StartException {
             b.append("Configured system properties:");
             for (String property : new TreeSet<String>(properties.stringPropertyNames())) {
                 String propVal = property.toLowerCase(Locale.getDefault()).contains("password") ? "<redacted>" : properties.getProperty(property, "<undefined>");
+                if (property.toLowerCase(Locale.getDefault()).equals("sun.java.command") && !propVal.isEmpty()) {
+                    Pattern pattern = Pattern.compile("(-D(?:[^ ])+=)((?:[^ ])+)");
+                    Matcher matcher = pattern.matcher(propVal);
+                    StringBuffer sb = new StringBuffer(propVal.length());
```

Comment:
Slight nitpick, this should probably be a `StringBuilder`. We shouldn't need a thread safe object here.

Reply:
appendReplacement(StringBuffer sb, String replacement) implementation is only for StringBuffer [1]. StringBuilder is available in Java 11 [2].
[1] https://docs.oracle.com/javase/8/docs/api/java/util/regex/Matcher.html
[2] https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/regex/Matcher.html

Reply:
Oops, my mistake. I was definitely looking at Java 11. Not a big deal at all :)